spec/core/data_structures.md

```diff
@@ -231,24 +226,46 @@ in the same round of the same height. Votes are lexicographically sorted on `Blo
 
 ```go
 type DuplicateVoteEvidence struct {
- VoteA  *Vote
- VoteB  *Vote
-
- Timestamp time.Time
+    VoteA  *Vote
+    VoteB  *Vote
 }
 ```
 
 Valid Duplicate Vote Evidence must adhere to the following rules:
 
-- Validator Address, Height, Round and Type of vote must be the same for both votes
+- Validator Address, Height, Round and Type must be the same for both votes
 
 - BlockID must be different for both votes (BlockID can be for a nil block)
 
 - Validator must have been in the validator set at that height
 
 - Vote signature must be valid (using the chainID)
 
-- Time must be equal to the block time
+- Evidence must not have expired: either age in terms of height or time must be
+    less than the age stated in the consensus params. Time is the block time that the
+    votes were a part of.
+
+### LightClientAttackEvidence
+
+```go
+type LightClientAttackEvidence struct {
+ ConflictingBlock *LightBlock
+ CommonHeight     int64
+}
+```
+
+Valid Light Client Attack Evidence encompasses three types of attack and must adhere to the following rules
+
+- If the header of the light block is invalid, thus indicating a lunatic attack, the node must check that
+    they can use `verifySkipping` from their header at the common height to the conflicting header
+
+- If the header is valid, then the validator sets are the same and this is either a form of equivocation
+    or amnesia. We therefore check that 2/3 of the validator set also signed the conflicting header
+
+- The trusted header of the node at the same height as the conflicting header must have a different hash to
+    the conflicting header.
```

Comment:
```suggestion
Valid Light Client Attack Evidence encompasses three types of attack and must adhere to the following rules

- The trusted header of the node at the same height as the conflicting header must have a different hash to the conflicting header.

- CommonHeight < ConflictingBlock.Header.Height 
    
- If CommonHeight + 1 = ConflictingBlock.Header.Height we do sequential verification (we check that 2/3 of the validator set also signed the conflicting header)

- If CommonHeight + 1 < ConflictingBlock.Header.Height we check `verifySkipping` 
```

I am not sure wether validity is the point around we should do the case distinction. I guess nodes may also do amnesia for invalid blocks?

Reply:
The actual implementation does in fact use the common height as the case distinction